app/scripts/lib/AccountIdentitiesPetnamesBridge.ts

```diff
@@ -4,44 +4,52 @@ import {
   NameType,
   NameOrigin,
 } from '@metamask/name-controller';
+import { InternalAccount } from '@metamask/keyring-api';
 import {
-  PreferencesController,
-  AccountIdentityEntry,
-} from '../controllers/preferences';
+  AccountsControllerChangeEvent,
+  AccountsControllerListAccountsAction,
+} from '@metamask/accounts-controller';
 import {
   PetnameEntry,
   AbstractPetnamesBridge,
   PetnamesBridgeMessenger,
 } from './AbstractPetnamesBridge';
 
+export type AccountIdentitiesPetnamesBridgeEvents =
+  AccountsControllerChangeEvent;
+export type AccountIdentitiesPetnamesBridgeActions =
+  AccountsControllerListAccountsAction;
 /**
  * A petnames bridge that uses the account identities from the preferences controller as the source.
  */
-export class AccountIdentitiesPetnamesBridge extends AbstractPetnamesBridge {
-  #preferencesController: PreferencesController;
-
+export class AccountIdentitiesPetnamesBridge extends AbstractPetnamesBridge<
+  AccountIdentitiesPetnamesBridgeEvents,
+  AccountIdentitiesPetnamesBridgeActions
+> {
   constructor({
-    preferencesController,
     nameController,
     messenger,
   }: {
-    preferencesController: PreferencesController;
     nameController: NameController;
-    messenger: PetnamesBridgeMessenger;
+    messenger: PetnamesBridgeMessenger<
+      AccountIdentitiesPetnamesBridgeEvents,
+      AccountIdentitiesPetnamesBridgeActions
+    >;
   }) {
     super({ isTwoWay: false, nameController, messenger });
-    this.#preferencesController = preferencesController;
   }
 
   /**
    * @override
    */
   protected getSourceEntries(): PetnameEntry[] {
-    const { identities } = this.#preferencesController.store.getState();
-    return Object.values(identities).map((identity: AccountIdentityEntry) => ({
-      value: identity.address,
+    const internalAccounts = this.messenger.call(
+      'AccountsController:listAccounts',
```

Comment:
Just confirming for audit sake this will still include all the standard account identities, but now also the snap account names?

Reply:
Yes, the accounts controller will have all the same accounts that the preference controller has.